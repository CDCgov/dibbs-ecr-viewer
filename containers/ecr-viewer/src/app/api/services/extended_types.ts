import {
  ColumnType,
  Generated,
  Insertable,
  Selectable,
  Updateable,
} from "kysely";

export interface ecr_data {
  eICR_ID: Generated<string>;
  set_id: string | undefined;
  fhir_reference_link: string | null;
  last_name: string | undefined;
  first_name: string | undefined;
  birth_date: ColumnType<Date | undefined>;
  gender: string | undefined;
  birth_sex: string | undefined;
  gender_identity: string | undefined;
  race: string | undefined;
  ethnicity: string | undefined;
  latitude: number | undefined;
  longitude: number | undefined;
  homelessness_status: string | undefined;
  disabilities: string | undefined;
  tribal_affiliation: string | undefined;
  tribal_enrollment_status: string | undefined;
  current_job_title: string | undefined;
  current_job_industry: string | undefined;
  usual_occupation: string | undefined;
  usual_industry: string | undefined;
  preferred_language: string | undefined;
  pregnancy_status: string | undefined;
  rr_id: string | undefined;
  processing_status: string | undefined;
  eicr_version_number: string | undefined;
  authoring_date: ColumnType<Date | undefined>;
  authoring_provider: string | undefined;
  provider_id: string | undefined;
  facility_id: string | undefined;
  facility_name: string | undefined;
  encounter_type: string | undefined;
  encounter_start_date: ColumnType<Date | undefined>;
  encounter_end_date: ColumnType<Date | undefined>;
  reason_for_visit: string | undefined;
  active_problems: string[] | undefined;
  date_created: ColumnType<Date | undefined>;
}

export interface patient_address {
  uuid: Generated<string>;
  use: "home" | "work" | "temp" | "old" | "billing" | undefined;
  type: "postal" | "physical" | "both" | undefined;
  text: string | undefined;
  line: string[] | undefined;
  city: string | undefined;
  district: string | undefined;
  state: string | undefined;
  postal_code: string | undefined;
  country: string | undefined;
  period_start: ColumnType<Date | undefined>;
  period_end: ColumnType<Date | undefined>;
  eICR_ID: string | null; // Nullable foreign key reference
}

export interface ecr_labs {
  uuid: Generated<string>;
  eICR_ID: string;
  test_type: string | undefined;
  test_type_code: string | undefined;
  test_type_system: string | undefined;
  test_result_qualitative: string | undefined;
  test_result_quantitative: number | null;
  test_result_units: string | undefined;
  test_result_code: string | undefined;
  test_result_code_display: string | undefined;
  test_result_code_system: string | undefined;
  test_result_interpretation: string | undefined;
  test_result_interpretation_code: string | undefined;
  test_result_interpretation_system: string | undefined;
  test_result_reference_range_low_value: number | string | null;
  test_result_reference_range_low_units: string | undefined;
  test_result_reference_range_high_value: number | string | null;
  test_result_reference_range_high_units: string | undefined;
  specimen_type: string | undefined;
  specimen_collection_date: ColumnType<Date | undefined>;
  performing_lab: string | undefined;
}

export interface ecr_rr_conditions {
  uuid: Generated<string>;
  eICR_ID: string;
  condition: string;
}

export interface ecr_rr_rule_summaries {
  uuid: Generated<string>;
  ecr_rr_conditions_id: string;
  rule_summary: string;
}

export type ExtendedECR = Selectable<ecr_data>;
export type NewExtendedECR = Insertable<ecr_data>;
export type ExtendedECRUpdate = Updateable<ecr_data>;

export type PatientAddress = Selectable<patient_address>;
export type NewPatientAddress = Insertable<patient_address>;
export type PatientAddressUpdate = Updateable<patient_address>;

export type ECRLabs = Selectable<ecr_labs>;
export type NewECRLabs = Insertable<ecr_labs>;
export type ECRLabsUpdate = Updateable<ecr_labs>;

export type ECRConditions = Selectable<ecr_rr_conditions>;
export type NewECRConditions = Insertable<ecr_rr_conditions>;
export type ECRConditionsUpdate = Updateable<ecr_rr_conditions>;

export type ECRRuleSummaries = Selectable<ecr_rr_rule_summaries>;
export type NewECRRuleSummaries = Insertable<ecr_rr_rule_summaries>;
export type ECRRuleSummariesUpdate = Updateable<ecr_rr_rule_summaries>;

export interface Extended {
  ecr_data: ecr_data;
  patient_address: patient_address;
  ecr_labs: ecr_labs;
  ecr_rr_conditions: ecr_rr_conditions;
  ecr_rr_rule_summaries: ecr_rr_rule_summaries;
}
