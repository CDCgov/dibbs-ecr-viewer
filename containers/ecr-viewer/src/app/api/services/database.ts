// Kysely ORM Connection Client

import { Kysely, PostgresDialect, MssqlDialect } from "kysely";
import { Pool } from "pg";
import * as tarn from "tarn";
import * as tedious from "tedious";

import { Extended } from "./extended_types";
import { Core } from "./types";

// Dialect to communicate with the database, interface to define its structure.

let db: Kysely<Core> | Kysely<Extended>;

if (process.env.METADATA_DATABASE_TYPE === "sqlserver") {
  db = new Kysely<Extended>({
    dialect: new MssqlDialect({
      tarn: {
        ...tarn,
        options: {
          min: 0,
          max: 100,
        },
      },
      tedious: {
        ...tedious,
        connectionFactory: () =>
          new tedious.Connection({
            authentication: {
              options: {
                password: process.env.SQL_SERVER_PASSWORD,
                userName: process.env.SQL_SERVER_USER || "sa",
              },
              type: "default",
            },
            options: {
              database: "master",
              port: 1433,
              trustServerCertificate: true,
              connectTimeout: 30000,
            },
            server: process.env.SQL_SERVER_HOST || "localhost",
          }),
      },
    }),
  });
} else {
  db = new Kysely<Core>({
    dialect: new PostgresDialect({
      pool: new Pool({
        database: process.env.POSTGRES_DATABASE || "ecr_viewer_db",
        host: process.env.POSTGRES_HOST,
        user: process.env.POSTGRES_USER,
        port: parseInt(process.env.POSTGRES_PORT || "5432"),
        max: parseInt(process.env.POSTGRES_MAX_THREADPOOL || "10"),
      }),
    }),
  });
}

export { db };
