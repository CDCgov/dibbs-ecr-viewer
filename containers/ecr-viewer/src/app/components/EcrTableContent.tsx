import React from "react";

import { listEcrData } from "@/app/services/listEcrDataService";
import { DateRangePeriod } from "@/app/utils/date-utils";

import { EcrTableDataRow } from "./EcrTableDataRow";

/**
 * eCR Table
 * @param props - The properties passed to the component.
 * @param props.currentPage - The current page to be displayed
 * @param props.itemsPerPage - The number of items to be displayed in the table
 * @param props.sortColumn - The column to sort by
 * @param props.sortDirection - The direction to sort by
 * @param props.filterDates - The date range used to filter data
 * @param props.searchTerm - The search term used to list data
 * @param props.filterConditions - (Optional) The reportable condition(s) used to filter the data
 * @returns - eCR Table element
 */
const EcrTableContent = async ({
  currentPage,
  itemsPerPage,
  sortColumn,
  sortDirection,
  searchTerm,
  filterConditions,
  filterDates,
}: {
  currentPage: number;
  itemsPerPage: number;
  sortColumn: string;
  sortDirection: string;
  filterDates: DateRangePeriod;
  searchTerm?: string;
  filterConditions?: string[];
}) => {
  const startIndex = (currentPage - 1) * itemsPerPage;

  const data = await listEcrData(
    startIndex,
    itemsPerPage,
    sortColumn,
    sortDirection,
    filterDates,
    searchTerm,
    filterConditions,
  );

  return (
    <tbody>
      {data.map((item, index) => (
        <EcrTableDataRow key={index} item={item} />
      ))}
    </tbody>
  );
};

export default EcrTableContent;
