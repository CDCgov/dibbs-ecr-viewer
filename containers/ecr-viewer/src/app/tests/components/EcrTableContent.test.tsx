import { axe } from "jest-axe";
import { act, render } from "@testing-library/react";
import EcrTableContent from "@/app/components/EcrTableContent";
import { EcrDisplay, listEcrData } from "@/app/services/listEcrDataService";
import router from "next-router-mock";

jest.mock("../../services/listEcrDataService");

jest.mock("next/navigation", () => ({
  useRouter: () => router,
  usePathname: () => router.pathname,
  useSearchParams: () => new URLSearchParams(router.asPath.split("?")[1] || ""),
}));

describe("EcrTableContent", () => {
  const mockedListEcrData = jest.mocked(listEcrData);
  const mockData: EcrDisplay[] = Array.from({ length: 25 }, (_, i) => ({
    ecrId: `id-${i + 1}`,
    patient_first_name: `first-${i + 1}`,
    patient_last_name: `last-${i + 1}`,
    dateModified: `2021-01-0${(i % 9) + 1}`,
    patient_date_of_birth: `2000-01-0${(i % 9) + 1}`,
    reportable_conditions: [
      `reportable-condition-${i + 1}`,
      `second-condition-${i + 1}`,
    ],
    rule_summaries: [`rule-summary-${i + 1}`, `second-summary-${i + 1}`],
    patient_report_date: i == 0 ? "" : `2021-01-0${(i % 9) + 1}`,
    date_created: `2021-01-0${(i % 9) + 1}`,
    eicr_set_id: `123${i}`,
    eicr_version_number: i == 0 ? undefined : `${i}`,
  }));
  const mockDateRange = {
    startDate: new Date("12-01-2024"),
    endDate: new Date("12-02-2024"),
  };

  beforeEach(() => {
    jest.resetAllMocks();
    router.setCurrentUrl(
      "/?itemsPerPage=25&columnId=report_date&direction=ASC&page=1",
    );
  });

  describe("load with an eCR", () => {
    it("should match snapshot", async () => {
      mockedListEcrData.mockResolvedValue(mockData);
      const table = document.createElement("table");
      const { container } = render(
        await EcrTableContent({
          currentPage: 1,
          itemsPerPage: 25,
          sortColumn: "date_created",
          sortDirection: "DESC",
          filterDates: mockDateRange,
        }),
        {
          container: document.body.appendChild(table),
        },
      );
      expect(container).toMatchSnapshot();
    });

    it("should pass accessibility", async () => {
      mockedListEcrData.mockResolvedValue(mockData);
      const table = document.createElement("table");
      const { container } = render(
        await EcrTableContent({
          currentPage: 1,
          itemsPerPage: 25,
          sortColumn: "date_created",
          sortDirection: "DESC",
          filterDates: mockDateRange,
        }),
        {
          container: document.body.appendChild(table),
        },
      );
      await act(async () => {
        expect(await axe(container)).toHaveNoViolations();
      });
    });
  });

  describe("load without an eCR", () => {
    it("should match snapshot", async () => {
      mockedListEcrData.mockResolvedValue([]);
      const table = document.createElement("table");
      const { container } = render(
        await EcrTableContent({
          currentPage: 1,
          itemsPerPage: 25,
          sortColumn: "date_created",
          sortDirection: "DESC",
          filterDates: mockDateRange,
        }),
        {
          container: document.body.appendChild(table),
        },
      );
      expect(container).toMatchSnapshot();
    });

    it("should pass accessibility", async () => {
      mockedListEcrData.mockResolvedValue([]);
      const table = document.createElement("table");
      const { container } = render(
        await EcrTableContent({
          currentPage: 1,
          itemsPerPage: 25,
          sortColumn: "date_created",
          sortDirection: "DESC",
          filterDates: mockDateRange,
        }),
        {
          container: document.body.appendChild(table),
        },
      );
      await act(async () => {
        expect(await axe(container)).toHaveNoViolations();
      });
    });
  });

  it("should call listEcrDataService with all params", async () => {
    mockedListEcrData.mockResolvedValue(mockData);

    const table = document.createElement("table");
    render(
      await EcrTableContent({
        currentPage: 1,
        itemsPerPage: 25,
        sortColumn: "date_created",
        sortDirection: "DESC",
        filterDates: mockDateRange,
        searchTerm: "blah",
        filterConditions: ["Anthrax (disorder)"],
      }),
      {
        container: document.body.appendChild(table),
      },
    );

    expect(mockedListEcrData).toHaveBeenCalledTimes(1);
    expect(mockedListEcrData).toHaveBeenCalledWith(
      0,
      25,
      "date_created",
      "DESC",
      mockDateRange,
      "blah",
      ["Anthrax (disorder)"],
    );
  });
});
