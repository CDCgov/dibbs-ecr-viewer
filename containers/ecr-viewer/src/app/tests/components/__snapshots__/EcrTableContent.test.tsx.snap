// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcrTableContent load with an eCR should match snapshot 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        <a
          href="/view-data?id=id-1"
        >
          First-1
           
          Last-1
        </a>
        <br />
        DOB: 
        2000-01-01
      </td>
      <td>
        2021-01-01
      </td>
      <td>
        <span
          class="no-data text-italic text-base"
        >
          No data
        </span>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-1
          </li>
          <li>
            second-condition-1
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-1
          </li>
          <li>
            second-summary-1
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-2"
        >
          First-2
           
          Last-2
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          1
        </span>
        <br />
        DOB: 
        2000-01-02
      </td>
      <td>
        2021-01-02
      </td>
      <td>
        2021-01-02
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-2
          </li>
          <li>
            second-condition-2
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-2
          </li>
          <li>
            second-summary-2
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-3"
        >
          First-3
           
          Last-3
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          2
        </span>
        <br />
        DOB: 
        2000-01-03
      </td>
      <td>
        2021-01-03
      </td>
      <td>
        2021-01-03
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-3
          </li>
          <li>
            second-condition-3
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-3
          </li>
          <li>
            second-summary-3
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-4"
        >
          First-4
           
          Last-4
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          3
        </span>
        <br />
        DOB: 
        2000-01-04
      </td>
      <td>
        2021-01-04
      </td>
      <td>
        2021-01-04
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-4
          </li>
          <li>
            second-condition-4
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-4
          </li>
          <li>
            second-summary-4
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-5"
        >
          First-5
           
          Last-5
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          4
        </span>
        <br />
        DOB: 
        2000-01-05
      </td>
      <td>
        2021-01-05
      </td>
      <td>
        2021-01-05
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-5
          </li>
          <li>
            second-condition-5
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-5
          </li>
          <li>
            second-summary-5
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-6"
        >
          First-6
           
          Last-6
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          5
        </span>
        <br />
        DOB: 
        2000-01-06
      </td>
      <td>
        2021-01-06
      </td>
      <td>
        2021-01-06
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-6
          </li>
          <li>
            second-condition-6
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-6
          </li>
          <li>
            second-summary-6
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-7"
        >
          First-7
           
          Last-7
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          6
        </span>
        <br />
        DOB: 
        2000-01-07
      </td>
      <td>
        2021-01-07
      </td>
      <td>
        2021-01-07
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-7
          </li>
          <li>
            second-condition-7
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-7
          </li>
          <li>
            second-summary-7
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-8"
        >
          First-8
           
          Last-8
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          7
        </span>
        <br />
        DOB: 
        2000-01-08
      </td>
      <td>
        2021-01-08
      </td>
      <td>
        2021-01-08
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-8
          </li>
          <li>
            second-condition-8
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-8
          </li>
          <li>
            second-summary-8
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-9"
        >
          First-9
           
          Last-9
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          8
        </span>
        <br />
        DOB: 
        2000-01-09
      </td>
      <td>
        2021-01-09
      </td>
      <td>
        2021-01-09
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-9
          </li>
          <li>
            second-condition-9
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-9
          </li>
          <li>
            second-summary-9
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-10"
        >
          First-10
           
          Last-10
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          9
        </span>
        <br />
        DOB: 
        2000-01-01
      </td>
      <td>
        2021-01-01
      </td>
      <td>
        2021-01-01
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-10
          </li>
          <li>
            second-condition-10
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-10
          </li>
          <li>
            second-summary-10
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-11"
        >
          First-11
           
          Last-11
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          10
        </span>
        <br />
        DOB: 
        2000-01-02
      </td>
      <td>
        2021-01-02
      </td>
      <td>
        2021-01-02
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-11
          </li>
          <li>
            second-condition-11
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-11
          </li>
          <li>
            second-summary-11
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-12"
        >
          First-12
           
          Last-12
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          11
        </span>
        <br />
        DOB: 
        2000-01-03
      </td>
      <td>
        2021-01-03
      </td>
      <td>
        2021-01-03
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-12
          </li>
          <li>
            second-condition-12
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-12
          </li>
          <li>
            second-summary-12
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-13"
        >
          First-13
           
          Last-13
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          12
        </span>
        <br />
        DOB: 
        2000-01-04
      </td>
      <td>
        2021-01-04
      </td>
      <td>
        2021-01-04
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-13
          </li>
          <li>
            second-condition-13
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-13
          </li>
          <li>
            second-summary-13
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-14"
        >
          First-14
           
          Last-14
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          13
        </span>
        <br />
        DOB: 
        2000-01-05
      </td>
      <td>
        2021-01-05
      </td>
      <td>
        2021-01-05
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-14
          </li>
          <li>
            second-condition-14
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-14
          </li>
          <li>
            second-summary-14
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-15"
        >
          First-15
           
          Last-15
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          14
        </span>
        <br />
        DOB: 
        2000-01-06
      </td>
      <td>
        2021-01-06
      </td>
      <td>
        2021-01-06
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-15
          </li>
          <li>
            second-condition-15
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-15
          </li>
          <li>
            second-summary-15
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-16"
        >
          First-16
           
          Last-16
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          15
        </span>
        <br />
        DOB: 
        2000-01-07
      </td>
      <td>
        2021-01-07
      </td>
      <td>
        2021-01-07
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-16
          </li>
          <li>
            second-condition-16
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-16
          </li>
          <li>
            second-summary-16
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-17"
        >
          First-17
           
          Last-17
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          16
        </span>
        <br />
        DOB: 
        2000-01-08
      </td>
      <td>
        2021-01-08
      </td>
      <td>
        2021-01-08
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-17
          </li>
          <li>
            second-condition-17
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-17
          </li>
          <li>
            second-summary-17
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-18"
        >
          First-18
           
          Last-18
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          17
        </span>
        <br />
        DOB: 
        2000-01-09
      </td>
      <td>
        2021-01-09
      </td>
      <td>
        2021-01-09
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-18
          </li>
          <li>
            second-condition-18
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-18
          </li>
          <li>
            second-summary-18
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-19"
        >
          First-19
           
          Last-19
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          18
        </span>
        <br />
        DOB: 
        2000-01-01
      </td>
      <td>
        2021-01-01
      </td>
      <td>
        2021-01-01
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-19
          </li>
          <li>
            second-condition-19
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-19
          </li>
          <li>
            second-summary-19
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-20"
        >
          First-20
           
          Last-20
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          19
        </span>
        <br />
        DOB: 
        2000-01-02
      </td>
      <td>
        2021-01-02
      </td>
      <td>
        2021-01-02
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-20
          </li>
          <li>
            second-condition-20
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-20
          </li>
          <li>
            second-summary-20
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-21"
        >
          First-21
           
          Last-21
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          20
        </span>
        <br />
        DOB: 
        2000-01-03
      </td>
      <td>
        2021-01-03
      </td>
      <td>
        2021-01-03
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-21
          </li>
          <li>
            second-condition-21
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-21
          </li>
          <li>
            second-summary-21
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-22"
        >
          First-22
           
          Last-22
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          21
        </span>
        <br />
        DOB: 
        2000-01-04
      </td>
      <td>
        2021-01-04
      </td>
      <td>
        2021-01-04
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-22
          </li>
          <li>
            second-condition-22
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-22
          </li>
          <li>
            second-summary-22
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-23"
        >
          First-23
           
          Last-23
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          22
        </span>
        <br />
        DOB: 
        2000-01-05
      </td>
      <td>
        2021-01-05
      </td>
      <td>
        2021-01-05
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-23
          </li>
          <li>
            second-condition-23
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-23
          </li>
          <li>
            second-summary-23
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-24"
        >
          First-24
           
          Last-24
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          23
        </span>
        <br />
        DOB: 
        2000-01-06
      </td>
      <td>
        2021-01-06
      </td>
      <td>
        2021-01-06
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-24
          </li>
          <li>
            second-condition-24
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-24
          </li>
          <li>
            second-summary-24
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <a
          href="/view-data?id=id-25"
        >
          First-25
           
          Last-25
        </a>
        <span
          class="usa-tag margin-x-1 padding-x-05 padding-y-2px bg-primary-lighter radius-md text-thin text-base-dark"
        >
          V
          24
        </span>
        <br />
        DOB: 
        2000-01-07
      </td>
      <td>
        2021-01-07
      </td>
      <td>
        2021-01-07
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            reportable-condition-25
          </li>
          <li>
            second-condition-25
          </li>
        </ul>
      </td>
      <td>
        <ul
          class="ecr-table-list"
        >
          <li>
            rule-summary-25
          </li>
          <li>
            second-summary-25
          </li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`EcrTableContent load without an eCR should match snapshot 1`] = `
<table>
  <tbody>
    <tr>
      <td
        class="text-middle text-center height-card"
        colspan="999"
      >
        <span
          class="text-bold font-body-lg"
        >
          No eCRs found. We couldn't find any eCRs matching your filter or search critera.
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;
